fem/fespace.cpp

```diff
@@ -2750,74 +2750,95 @@ L2ProjectionGridTransfer::L2Projection::L2Projection(
      fes_lor(fes_lor_)
 {
    Mesh *mesh_ho = fes_ho.GetMesh();
-   MFEM_VERIFY(mesh_ho->GetNumGeometries(mesh_ho->Dimension()) <= 1,
-               "mixed meshes are not supported");
+   Mesh *mesh_lor = fes_lor.GetMesh();
+   int nel_ho = mesh_ho->GetNE();
+   int nel_lor = mesh_lor->GetNE();
 
    // If the local mesh is empty, skip all computations
-   if (mesh_ho->GetNE() == 0) { return; }
+   if (nel_ho == 0) { return; }
 
-   const FiniteElement *fe_lor = fes_lor.GetFE(0);
-   const FiniteElement *fe_ho = fes_ho.GetFE(0);
-   ndof_lor = fe_lor->GetDof();
-   ndof_ho = fe_ho->GetDof();
+   const CoarseFineTransformations &cf_tr = mesh_lor->GetRefinementTransforms();
 
-   const int nel_lor = fes_lor.GetNE();
-   const int nel_ho = fes_ho.GetNE();
-
-   nref = nel_lor/nel_ho;
+   int nref_max = 0;
+   Array<Geometry::Type> geoms;
+   mesh_ho->GetGeometries(mesh_ho->Dimension(), geoms);
+   for (int ig=0; ig<geoms.Size(); ++ig)
+   {
+      Geometry::Type geom = geoms[ig];
+      nref_max = std::max(nref_max, cf_tr.point_matrices[geom].SizeK());
+   }
 
    // Construct the mapping from HO to LOR
    // ho2lor.GetRow(iho) will give all the LOR elements contained in iho
-   ho2lor.SetSize(nel_ho, nref);
-   const CoarseFineTransformations &cf_tr =
-      fes_lor.GetMesh()->GetRefinementTransforms();
+   ho2lor.SetSize(nel_ho, nref_max);
    for (int ilor=0; ilor<nel_lor; ++ilor)
    {
       int iho = cf_tr.embeddings[ilor].parent;
       ho2lor.AddConnection(iho, ilor);
    }
    ho2lor.ShiftUpI();
 
+   offsets.SetSize(nel_ho+1);
+   offsets[0] = 0;
+   for (int iho=0; iho<nel_ho; ++iho)
```

Comment:
I can't find anything substantial so I'm nitpicking: the whitespace here and there could be unified, as in `for (int iho = 0; iho < nel_ho; ++iho)`, it's less crammed and generally preferred this way in MFEM...

Reply:
Sure, I changed these.